frontend/src/pages/appointmentsettings/AppointmentStepsWindow.tsx

```diff
@@ -5,28 +5,112 @@ import Typography from "@mui/material/Typography";
 import Box from "@mui/material/Box";
 import Button from "@mui/material/Button";
 import AppointmentStepsPages from "./AppointmentStepsPages";
+import {useNavigate} from "react-router-dom";
+import {useContext, useState} from "react";
+import {dataPickedTime} from "../../models/appointmentsettings/DataPickedTime";
+import {dataPickedRestaurant} from "../../models/appointmentsettings/DataPickedRestaurant";
+import {DataCompleteDateInfos, transferSettingsToDB} from "../../service/tastydate-api-service";
+import {AuthContext} from "../../context/AuthProvider";
+import { v4 as uuidv4 } from 'uuid';
 
 interface AppointmentStepsWindowProps {
     activeStep: number,
+    setActiveStep: Function,
     steps: string[],
     isStepOptional: Function,
-    handleReset: Function,
+    handleEdit: Function,
     handleBack: Function,
     handleSkip: Function,
-    handleNext: Function
+    handleNext: Function,
+    setTransferSettingsItem: Function
 }
 
 export default function AppointmentStepsWindow(
     {
         activeStep,
+        setActiveStep,
         steps,
         isStepOptional,
-        handleReset,
+        handleEdit,
         handleBack,
         handleSkip,
-        handleNext
+        handleNext,
+        setTransferSettingsItem
     }: AppointmentStepsWindowProps) {
 
+    const {token}=useContext(AuthContext)
+
+    const navigate = useNavigate();
+
+    //AppointOne
+    const [tastyDateName, setTastyDateName] = useState("");
+    const [location, setLocation] = useState("");
+    const [notes, setNotes] = useState("");
+    const [chosenDisplayName, setChosenDisplayName] = useState("");
+
+    //AppointTwo
+    const [date, setDate] = useState<Date | null>(null);
+    const [startTime, setStartTime] = useState<Date | null>(null);
+    const [endTime, setEndTime] = useState<Date | null>(null);
+    const [idPickedTime, setIdPickedTime] = useState<number>(1);
+    const [dataDateTimes, setDataDateTimes] = useState<dataPickedTime[]>([]);
+
+    //AppointThree
+    const [category, setCategory] = useState<string>("");
+    const [postcode, setPostcode] = useState<number>(1);
+    const [city, setCity] = useState<string>("");
+    const [restaurantName, setRestaurantName] = useState<string>("");
+    const [rating, setRating] = useState<number>(0);
+    const [price, setPrice] = useState<number>(0);
+    const [idPickedRestaurant, setIdPickedRestaurant] = useState<number>(1);
+    const [restaurantData, setRestaurantData] = useState<dataPickedRestaurant[]>([]);
```

Comment:
I would extract all these state into a custom hook, and return the values, setter and maybe a delete function.

Reply:
This helped me a lot! I already thought of custom hooks. Thanks for your explanation. I have successfully implemented it in my code now.